src/java.base/share/classes/java/security/KeyStore.java

```diff
@@ -2098,58 +2087,53 @@ public KeyStore run() throws Exception {
                             }
                         }
                     }
-                    public KeyStore run0() throws Exception {
-                        KeyStore ks;
-                        char[] password;
+                } catch (Exception e) {
+                    oldException = e;
```

Comment:
The updated KeyStore.java code looks good. The run0() is replaced by getKeyStore0(). Here it does not do e.getCause(), and it looks like because the exception is not wrapped by PrivilegedExceptionAction anymore?

Reply:
Yes, it now throws the exception directly so there is no need to extract the cause.

Also, I fixed an infinite loop issue which was causing a couple of regression tests to fail. See https://github.com/openjdk/jdk/pull/22401/commits/9ee1967c7883355c4ef9ca20d8e3926ad34914e2

Reply:
Thanks for the confirmation. Updated webrev looks good.